### DISCUSSION TITLE
some nodes rendered displayed incomplete

### DISCUSSION THREAD

**Discussion by edst42** [mapsforge/mapsforge#1210] [Q&A] [answered]
Hi,
I am using Mapsforge with a custom created map. Based on .osm file a .map is created with osmosis. The .osm file contains city node definitions, of which the labels are referenced in the rendertheme.xml file.
This works nicely except that some nodes are not displayed completely. See picture below. It seems they are overlayed partly with map tiles. I have no idea however how I can prevent this. The order of definitions in the rendertheme file does not seem to influence it. What could I be doing wrong?

![cityNodes](https://user-images.githubusercontent.com/26207749/117553527-7021c400-b052-11eb-979c-9898e5708444.png)

**Comment by devemux86** [comment, marked as answer]
Do you use the Mapsforge library API? These look cut at tile bounds.

Mapsforge renderer should work fine. See the [getting started](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/GettingStarted.java) example.

Or you can use a label layer for better symbol / label rendering.
See the [documentation](https://github.com/mapsforge/mapsforge/blob/master/docs/LabelLayer.md) and the label layer examples in [mapsforge-samples-android](https://github.com/mapsforge/mapsforge/tree/master/mapsforge-samples-android).

**Comment by edst42** [comment]
I am using Mapsforge indeed, but only the TilerRendererLayer, not the LabelLayer. I will try if the latter helps. Thanks!
